src/db/gc/mod.rs

```diff
@@ -473,4 +477,43 @@ mod test {
             current_epoch + 1 + depth * 2
         );
     }
+
+    #[tokio::test]
+    async fn persistent_data_resilient_to_gc() {
+        let depth = 5 as ChainEpochDelta;
+        let current_epoch = 0 as ChainEpochDelta;
+
+        let tester = GCTester::new();
```

Comment:
Shall we test both `MemoryDB` and `ParityDB`?

Reply:
ParityDB is tricky to test, there is an issue for that. Therefore it has been tested manually.

Reply:
The main issue there is that deletes are not propagating immediately, so unit testing it reliably is not currently possible.

Reply:
actually, no issue. creating one